api.go

```diff
@@ -528,6 +532,11 @@ func (c Client) executeMethod(method string, metadata requestMetadata) (res *htt
 		// For all other cases break out of the retry loop.
 		break
 	}
+
+	if retryCount >= MaxRetry {
+		err = errors.New("Max retry reached")
```

Comment:
I think it's better to declare the error once at the package level, and return that instead.  

On the other hand, in this case, would you want to return the last error received?  I was thinking if you just return "Max retry reached", the caller won't have any idea what the underlying error was.